### ISSUE TITLE
Bug when working with different AudioContexts in WebAudio mode

### ISSUE THREAD

**Issue by phal0r** [edimuj/cordova-plugin-audioinput#51]
Hi @edimuj ,

thanks for this nice plugin. While testing I discovered a minor bug, that lead to the crash of my webapp running in cordova under iOS. I am using webaudio and your plugin to record audio and to prevent blocking hardware resources I close and release the audio context once the recording is finished. Subsequent calls to audioinput.start would fail with a strange error.

How to reproduce:

call to start recording
```
audioinput.start({
	streamToWebAudio: true,
	channels: audioinput.CHANNELS.STEREO,
	audioContext: recordingAudioContext,
	bufferSize: bufferSize
});
```
connect the audioinput to my audio graph and once finished i call 
```
audioinput.stop();
```

The first call is being handled successfully, but the next calls, as mentioned before, will fail. I create a new recordingAudioContext on every new tap on the record button of the user. The problem is, that you create a gainnode as the connecting node to the rest of the audio graph, which is totally fine. But while calling audioinput.stop() the gainnode should eventually be discarded, since it is possible to pass a new audioContext on every call of audioinput.start.

Long story short:
- the gainnode is created during the first call of start() with the provided audioContext
- I pass a new audioContext on every subsequent call of start(), but the plugin wants to use the "old" gainnode, which belongs the old already closed audioContext, which fails with an exception

My current workaround:

manually discard the gainnode via `audioinput._micGainNode = null;`, so a new one is created automatically thus resolving the issue, but I guess the plugin should handle this internally :)

**Comment by edimuj** [comment]
Thanks, @phal0r for reporting this and for the thourogh explanation. I´ll look into this as sson as I can, and provide a fix. If you want it sooner than later, feel free to submit a PR.

**Comment by phal0r** [comment]
Thanks for coming back to this bug :)

**Comment by edimuj** [comment]
And thank you for the thorough explanation of the error, made it simple for me to fix the issue.

**Comment by kpwa** [comment]
i am experiencing this problem in ios 11. i understand an update is applied for tis fix but in ios its still happening
